src/admin/__tests__/fetchOffsets.spec.js

```diff
@@ -64,5 +72,221 @@ describe('Admin', () => {
+          groupId,
+          topic: topicName,
+        })
+
+        expect(offsetsBeforeResolving).toEqual([{ partition: 0, offset: '-1', metadata: null }])
+        expect(offsetsUponResolving).toEqual([{ partition: 0, offset: '10', metadata: null }])
+        expect(offsetsAfterResolving).toEqual([{ partition: 0, offset: '10', metadata: null }])
+      })
+
+      test('reset to earliest: returns earliest *topic* offsets after resolving', async () => {
+        await admin.resetOffsets({ groupId, topic: topicName, earliest: true })
+
+        const offsetsBeforeResolving = await admin.fetchOffsets({
+          groupId,
+          topic: topicName,
+        })
+        const offsetsUponResolving = await admin.fetchOffsets({
+          groupId,
+          topic: topicName,
+          resolveOffsets: true,
+        })
+        const offsetsAfterResolving = await admin.fetchOffsets({
+          groupId,
+          topic: topicName,
+        })
+
+        expect(offsetsBeforeResolving).toEqual([{ partition: 0, offset: '-2', metadata: null }])
+        expect(offsetsUponResolving).toEqual([{ partition: 0, offset: '0', metadata: null }])
+        expect(offsetsAfterResolving).toEqual([{ partition: 0, offset: '0', metadata: null }])
+      })
+
+      testIfKafkaAtLeast_0_11(
+        'will return the correct earliest offset when it is greater than 0',
+        async () => {
+          // simulate earliest offset = 7, by deleting first 7 messages from the topic
+          const messagesToDelete = [
+            {
+              topic: topicName,
+              partitions: [
+                {
+                  partition: 0,
+                  offset: '7',
+                },
+              ],
+            },
+          ]
+
+          await deleteRecords({ cluster, topicName, topicDetails: messagesToDelete })
+          await admin.resetOffsets({ groupId, topic: topicName, earliest: true })
+
+          const offsetsBeforeResolving = await admin.fetchOffsets({
+            groupId,
+            topic: topicName,
+          })
+          const offsetsUponResolving = await admin.fetchOffsets({
+            groupId,
+            topic: topicName,
+            resolveOffsets: true,
+          })
+          const offsetsAfterResolving = await admin.fetchOffsets({
+            groupId,
+            topic: topicName,
+          })
+
+          expect(offsetsBeforeResolving).toEqual([{ partition: 0, offset: '-2', metadata: null }])
+          expect(offsetsUponResolving).toEqual([{ partition: 0, offset: '7', metadata: null }])
+          expect(offsetsAfterResolving).toEqual([{ partition: 0, offset: '7', metadata: null }])
+        }
+      )
+    })
   })
 })
+
+/**
+ * Manually delete records up to the selected offset
+ * http://kafka.apache.org/protocol.html#The_Messages_DeleteRecords
+ * Only available from Kafka 0.11 upwards
+ */
+const deleteRecords = async ({ cluster, topicName, topicDetails }) => {
```

Comment:
I definitely think we should move this into the supported protocols rather than having it here, but we can certainly do that in a separate PR rather than bloating this one.